include/translated-object.php

```diff
@@ -127,44 +129,68 @@ public function update_language( $id, $lang ) {
 	public function get_object_term( $object_id, $taxonomy ) {
-		if ( empty( $object_id ) || is_wp_error( $object_id ) ) {
+		$object_id = $this->sanitize_int_id( $object_id );
+
+		if ( empty( $object_id ) ) {
 			return false;
 		}
 
-		$object_id = (int) $object_id;
+		$term = get_object_term_cache( $object_id, $taxonomy );
 
-		if ( $object_id < 0 ) {
-			return false;
+		if ( is_array( $term ) ) {
+			return ! empty( $term ) ? reset( $term ) : false;
 		}
 
-		$term = get_object_term_cache( $object_id, $taxonomy );
+		// Query language and translations at the same time.
+		$taxonomies = array( $this->tax_language, $this->tax_translations );
 
-		if ( false === $term ) {
-			// Query language and translations at the same time.
-			$taxonomies = array( $this->tax_language, $this->tax_translations );
-
-			// Query terms.
-			$terms = array();
-			$object_terms = wp_get_object_terms( $object_id, $taxonomies, array( 'update_term_meta_cache' => false ) );
-			if ( is_array( $object_terms ) ) {
-				foreach ( $object_terms as $t ) {
-					$terms[ $t->taxonomy ] = $t;
-					if ( $t->taxonomy == $taxonomy ) {
-						$term = $t;
-					}
+		// Query terms.
+		$terms        = array();
+		$term         = false;
+		$object_terms = wp_get_object_terms( $object_id, $taxonomies, array( 'update_term_meta_cache' => false ) );
+
+		if ( is_array( $object_terms ) ) {
+			foreach ( $object_terms as $t ) {
+				$terms[ $t->taxonomy ] = $t;
+				if ( $t->taxonomy === $taxonomy ) {
+					$term = $t;
 				}
 			}
+		}
 
-			// Stores it the way WP expects it. Set an empty cache if no term was found in the taxonomy.
-			foreach ( $taxonomies as $tax ) {
-				wp_cache_add( $object_id, empty( $terms[ $tax ] ) ? array() : array( $terms[ $tax ] ), $tax . '_relationships' );
-			}
+		// Stores it the way WP expects it. Set an empty cache if no term was found in the taxonomy.
+		foreach ( $taxonomies as $tax ) {
+			wp_cache_add( $object_id, empty( $terms[ $tax ] ) ? array() : array( $terms[ $tax ] ), $tax . '_relationships' );
 		}
-		else {
-			$term = reset( $term );
+
+		return $term;
+	}
+
+	/**
+	 * Returns a list of post translations, given a `tax_translations` term ID.
+	 *
+	 * @since 3.2
+	 *
+	 * @param  int $term_id Term ID.
+	 * @return array<int>   An associative array of translations with language code as key and translation id as value.
+	 */
+	public function get_translations_from_term_id( $term_id ) {
+		$term_id = $this->sanitize_int_id( $term_id );
+
+		if ( empty( $term_id ) ) {
+			return array();
 		}
 
-		return empty( $term ) ? false : $term;
+		$translations_term = get_term( $term_id, $this->tax_translations );
+
+		if ( ! $translations_term instanceof WP_Term || empty( $translations_term->description ) ) {
+			return array();
+		}
+
+		// Lang slugs as array keys, template IDs as array values.
+		$translations = maybe_unserialize( $translations_term->description );
+
+		return $this->validate_translations( $translations, $term_id, 'display' );
```

Comment:
I would rather pass 0 instead of `$term_id` which looks incorrect to me.
Coud you add tests to cover this new method?

Reply:
Fixed by [0a2d22d](https://github.com/polylang/polylang/pull/984/commits/0a2d22dcff92cf13990cd27e25c47b121c3225bb).
Test added by [6aaa8c2](https://github.com/polylang/polylang/pull/984/commits/6aaa8c2a75a6eac1b3cff6873776015f02f9f834).